gordo_components/client/forwarders.py

```diff
@@ -152,39 +156,58 @@ def __init__(
             if len(sub_df.columns) == len(endpoint.tag_list):
                 sub_df.columns = [tag.name for tag in endpoint.tag_list]
 
-            # Async write predictions to influx
-            with ThreadPoolExecutor(max_workers=1) as executor:
-                logger.info(
-                    f"Writing {len(sub_df)} points to Influx for measurement: {top_lvl_name}"
-                )
-                future = executor.submit(
-                    self.dataframe_client.write_points,
-                    dataframe=sub_df,
-                    measurement=top_lvl_name,
-                    tags=tags,
-                    batch_size=10000,
-                )
-                await asyncio.wrap_future(future)
+            await self._write_to_influx_with_retries(sub_df, tags, top_lvl_name)
+
+    async def _write_to_influx_with_retries(self, df, tags, measurement):
+        """Async write data to influx with retries and exponential backof. Will sleep
+        exponentially longer between each retry, starting at 8 seconds, capped at 5 min.
+        """
+
+        with ThreadPoolExecutor(max_workers=1) as executor:
+            logger.info(
+                f"Writing {len(df)} points to Influx for measurement: {measurement}"
+            )
+            for current_attempt in itertools.count(start=1):
+                try:
+                    future = executor.submit(
+                        self.dataframe_client.write_points,
+                        dataframe=df,
+                        measurement=measurement,
+                        tags=tags,
+                        batch_size=10000,
+                    )
+                    await asyncio.wrap_future(future)
+                except Exception as exc:
+                    if current_attempt <= self.n_retries:
+                        # Sleep at most 5 min
+                        time_to_sleep = min(2 ^ (current_attempt + 2), 300)
+                        logger.warning(
+                            f"Failed to forward data to influx on attempt "
+                            f"{current_attempt} out of {self.n_retries} attempts. "
+                            f"Error: {exc}."
+                            f"Sleeping {time_to_sleep} seconds and trying again"
+                        )
+                        time.sleep(time_to_sleep)  # Not async on purpose.
```

Comment:
I missed this review entirely, but this will end up holding up _all_ other coroutines in the current batch, no? Why was async sleep not an option.